orchagent/portsorch.cpp

```diff
@@ -2264,26 +2261,41 @@ void PortsOrch::doPortTask(Consumer &consumer)
             SWSS_LOG_NOTICE("Deleting Port %s", alias.c_str());
             auto port_id = m_portList[alias].m_port_id;
             auto hif_id = m_portList[alias].m_hif_id;
+            auto bridge_port_oid = m_portList[alias].m_bridge_port_id;
 
-            if (m_portList[alias].has_dependency())
+            if (bridge_port_oid != SAI_NULL_OBJECT_ID)
             {
-                // Port has one or more dependencies, cannot remove
-                SWSS_LOG_WARN("Cannot to remove port because of dependency");
+                // Bridge port OID is set on a port as long as
+                // port is part of at-least one VLAN. 
```

Comment:
>VLAN [](start = 48, length = 4)

how about vxlan?

Reply:
> > bool PortsOrch::removePort(sai_object_id_t port_id)
> 
> The function behavior is weird:
> 
> 1. if removed, return true
> 2. if in use, return false
> 3. if other error, throw.
> 
> Suggest return an enum, or sai_status_t. No throw.
> 
> Refers to: orchagent/portsorch.cpp:1569 in [ca4812c](https://github.com/Azure/sonic-swss/commit/ca4812c9105f60d15a85903648b4935ad514b700). [](commit_id = [ca4812c](https://github.com/Azure/sonic-swss/commit/ca4812c9105f60d15a85903648b4935ad514b700), deletion_comment = False)

Half heartedly, I agree. I am NOT too happy because, we always handle the sai_status in the inner most function (where we call sai API's) and return only true of false to the calling function. And its pretty consistent throughout the code and I liked it. But here I am making an exception by returning sai_status to the caller. I did think about introducing our own enum, but that seemed like over kill. So, I will stick to this for now.

Reply:
> > VLAN [](start = 48, length = 4)
> 
> how about vxlan?

For VxLAN I do NOT see any direct dependency on physical port. If your question is more about handling tunnel interface dependency on physical port, that is NOT in scope of this PR. Because, we expect every dependency to be tracked in SAI REDIS layer. Please let me know if I dint interpret your comment correctly